src/m365/context/commands/context-remove.spec.ts

```diff
@@ -0,0 +1,176 @@
+    sinon.stub(fs, 'readFileSync').callsFake(_ => JSON.stringify({
+      "context": {}
+    }));
+    const unlinkSyncStub = sinon.stub(fs, 'unlinkSync').callsFake(_ => { });
+    await command.action(logger, { options: { debug: true, confirm: true } });
+
+    assert(unlinkSyncStub.called);
+  });
+
+  it(`removes the context info from the existing .m365rc.json file`, async () => {
+    let fileContents: string | undefined;
+    let filePath: string | undefined;
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
+      { continue: true }
+    ));
+
+    sinon.stub(fs, 'existsSync').callsFake(_ => true);
+    sinon.stub(fs, 'readFileSync').callsFake(_ => JSON.stringify({
+      "apps": [
+        {
+          "appId": "e23d235c-fcdf-45d1-ac5f-24ab2ee0695d",
+          "name": "CLI app"
+        }
+      ],
+      "context": {}
+    }));
+    sinon.stub(fs, 'writeFileSync').callsFake((_, contents) => {
+      filePath = _.toString();
+      fileContents = contents as string;
+    });
+
+    await command.action(logger, { options: { debug: true } });
+
+    assert.strictEqual(filePath, '.m365rc.json');
+    assert.strictEqual(fileContents, JSON.stringify({
+      apps: [
+        {
+          "appId": "e23d235c-fcdf-45d1-ac5f-24ab2ee0695d",
+          "name": "CLI app"
+        }
+      ]
+    }, null, 2));
+  });
+
+  it(`doesn't save context info in the .m365rc.json file when there was an error reading file contents`, async () => {
+    sinon.stub(fs, 'existsSync').callsFake(_ => true);
+    sinon.stub(fs, 'readFileSync').callsFake(_ => { throw new Error('An error has occurred'); });
+    const fsWriteFileSyncSpy = sinon.spy(fs, 'unlinkSync');
+
+    await command.action(logger, { options: { debug: true, confirm: true } });
+    assert(fsWriteFileSyncSpy.notCalled);
+  });
+
+  it(`doesn't save context info in the .m365rc.json file when there was error writing file contents`, async () => {
+    sinon.stub(fs, 'existsSync').callsFake(_ => true);
+    sinon.stub(fs, 'readFileSync').callsFake(_ => JSON.stringify({
+      "apps": [
+        {
+          "appId": "e23d235c-fcdf-45d1-ac5f-24ab2ee0695d",
+          "name": "CLI app"
+        }
+      ],
+      "context": {}
+    }));
+    sinon.stub(fs, 'writeFileSync').callsFake(_ => { throw new Error('An error has occurred'); });
+    command.action(logger, { options: { debug: true, confirm: true } });
+  });
+
+  it(`doesn't remove the .m365rc.json file when there was error removing the file`, async () => {
+    sinon.stub(fs, 'existsSync').callsFake(_ => true);
+    sinon.stub(fs, 'readFileSync').callsFake(_ => JSON.stringify({
+      "context": {}
+    }));
+    sinon.stub(fs, 'unlinkSync').callsFake(_ => { throw new Error('An error has occurred'); });
+    command.action(logger, { options: { debug: true, confirm: true } });
+  });
+
+  it(`doesn't remove context info from the .m365rc.json file when there was no context`, async () => {
```

Comment:
Let's rephrase the test name so that it's a better description of what we do, eg: `doesn't update the context file, if it doesn't contain context information`